src/apis/stockx/index.js

```diff
@@ -0,0 +1,277 @@
+
+import { CookieJar } from 'tough-cookie';
+
+// APIs
+import ProductsApi from './products';
+import AsksApi from './asks';
+import BidsApi from './bids';
+import UserApi from './user';
+
+// Utils
+import { format, currencies } from '../utils';
+
+export default class StockX {
+    /**
+     * 
+     * @param {Object} options
+     * @param options.proxy - The proxy to use for future requests
+     * @param options.currency - The currency used when making requests
+     */
+    constructor(options = {}){
+        const { proxy = null, currency = currencies.USD } = options;
+
+        //Configure options
+        this.currency = currency;
+        this.cookieJar = new CookieJar();
+        this.proxy = proxy ? format(proxy) : null;
+
+        this.context = {
+            currency: this.currency,
+            jar: this.cookieJar,
+            baseURL: 'https://stockx.com/api',
+            headers: {
+                origin: 'https://accounts.stockx.com',
+                host: 'accounts.stockx.com',
+                'sec-fetch-mode': 'cors',
+                'accept-language': 'en-US,en;q=0.9',
+                'x-requested-with': 'XMLHttpRequest',
+                appos: 'web',
+                'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36',
+                accept: '*/*',
+                authority: 'stockx.com',
+                'sec-fetch-site': 'same-origin',
+                appversion: '0.1',
+                'content-type': 'application/json',
+            },
+            proxy: this.proxy,
+        };
+
+        // sub-module constructors
+        this.products = new ProductsApi(this.context);
+        this.asks = new AsksApi(this.context);
+        this.bids = new BidsApi(this.context);
+        this.user = new UserApi(this.context);
+    };
+
```

Comment:
Will want to delete all this once we make sure everything is tested and ready to go.